tools/generators/files_and_groups/src/Generator/CalculatePathTree.swift

```diff
@@ -1,97 +1,183 @@
 import PBXProj
 
 extension Generator {
-    static func calculatePathTree(paths: Set<BazelPath>) -> PathTreeNode {
+    static func calculatePathTree(
+        paths: [BazelPath],
+        generatedPaths: [GeneratedPath]
+    ) -> [PathTreeNode] {
+        /// `[package: [config: [path]]`
+        var generatedPathsByPackageAndConfig:
+            [BazelPath: [String: [BazelPath]]] = [:]
+        for generatedPath in generatedPaths {
+            generatedPathsByPackageAndConfig[
+                generatedPath.package,
+                default: [:]
+            ][generatedPath.config, default: []].append(generatedPath.path)
+        }
+
+        // FIXME: Do this in parallel
```

Comment:
Is this still relevant/are you looking to parallelize the for loop? For reference, I did some basic benchmarking and found this branch the same and even marginally faster.

Reply:
Yeah, I'll still probably do this is parallel. Just as one of the last FIXME. And I might do